program/src/instruction.rs

```diff
@@ -170,6 +170,33 @@ pub enum TokenWrapInstruction {
     /// 8. `[]` (Optional) Owner program. Required when metadata account is
     ///    owned by a third-party program.
     SyncMetadataToSplToken,
+
+    /// Creates or updates the canonical program pointer for a mint.
+    ///
+    /// The mint authority of an unwrapped mint may desire to deploy a forked
+    /// version of the Token Wrap program themselves. It's likely the case they
+    /// prefer a certain set of extensions or a particular config for the
+    /// wrapped token-2022s. They may even freeze the unwrapped mint's
+    /// escrow account in the original deployment to force the use of the fork.
+    /// A `CanonicalPointer` PDA allows a mint authority to signal on-chain
+    /// another Token Wrap deployment is the "canonical" one for the mint.
+    ///
+    /// If calling for the first time, the client is responsible for pre-funding
+    /// the rent for the PDA that will be initialized.
+    ///
+    /// If no mint authority exists on the unwrapped mint, this instruction will
+    /// fail.
+    ///
+    /// Accounts expected:
+    /// 0. `[s]` Unwrapped mint authority
```

Comment:
Did we confirm that this is ok from the foundation? If an SPL token mint is using an SPL token multisig as the authority, we can't actually get a signature for it 😅

USDC and USDT both use SPL token multisigs as the mint *and* freeze authority, so we might need to add SPL multisig support to this instruction, which may require rejiggering the order of accounts here.

Separately, what if we allowed for a signature from either the freeze authority or the mint authority?

Reply:
> Did we confirm that this is ok from the foundation?

Are you talking about the feature generally? In NY, they specifically requested a way for a mint authority to signal another token-wrap deployment was the canonical one for the mint.

> If an SPL token mint is using an SPL token multisig as the authority, we can't actually get a signature for it 😅

Oh shoot. Totally forgot that multisigs cannot directly sign txs. Yes, I'll need to reorder the accounts for that usecase like the other ixs.

> Separately, what if we allowed for a signature from either the freeze authority or the mint authority?

Yes, that sounds reasonable too.